llvm/include/llvm/IR/DebugLoc.h

```diff
@@ -48,22 +64,29 @@ namespace llvm {
     Temporary
   };
 
-  // Extends TrackingMDNodeRef to also store a DebugLocKind, allowing Debugify
-  // to ignore intentionally-empty DebugLocs.
-  class DILocAndCoverageTracking : public TrackingMDNodeRef {
+  // Extends TrackingMDNodeRef to also store a DebugLocKind and Origin,
+  // allowing Debugify to ignore intentionally-empty DebugLocs and display the
+  // code responsible for generating unintentionally-empty DebugLocs.
+  // Currently we only need to track the Origin of this DILoc when using a
+  // DebugLoc that is Normal and empty, so only collect the origin stacktrace in
```

Comment:
Do you "normal or empty"...? There seem to be two cases below where traces are collected.

stacktrace -> stack trace

Reply:
Ah okay, re-reading it again, I think it makes sense this time. Not quite sure how to best clarify for other readers... Maybe move some of your "Normal" explanation here into the code comment...?